### ISSUE TITLE
No feedback on error

### ISSUE THREAD

**Issue by hadess** [vinszent/gnome-twitch#218]
```
Dec 06 10:51:34 classic com.vinszent.GnomeTwitch.desktop[29731]: [09:51:34] Warning - GNOME-Twitch : {GtTwitch} Received unsuccessful response from url 'https://api.twitch.tv/kraken/streams?limit=30&offset=0&game=' with code '2' and body '(null)'
Dec 06 10:51:34 classic com.vinszent.GnomeTwitch.desktop[29731]: [09:51:34] Warning - GNOME-Twitch : {GtTwitch} Error sending message to get top channels
Dec 06 10:51:34 classic com.vinszent.GnomeTwitch.desktop[29731]: [09:51:34] Warning - GNOME-Twitch : {GtTwitch} Received unsuccessful response from url 'https://api.twitch.tv/kraken/games/top?limit=30&offset=0' with code '2' and body '(null)'
Dec 06 10:51:34 classic com.vinszent.GnomeTwitch.desktop[29731]: [09:51:34] Warning - GNOME-Twitch : {GtTwitch} Error sending message to get top channels
Dec 06 10:53:56 classic com.vinszent.GnomeTwitch.desktop[29731]: [09:53:56] Critical - GLib-GObject : g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Dec 06 10:53:56 classic com.vinszent.GnomeTwitch.desktop[29731]: [09:53:56] Critical - GLib-GObject : g_boxed_free: assertion 'boxed != NULL' failed
```

When this fails (and I don't know why it does), there should really have a better empty state:
https://wiki.gnome.org/Design/OS/EmptyStates

I can't seem to find the one with the "Oops" symbolic icon, that would say that the games list couldn't be loaded.

I used TingPing's flatpak build.

**Comment by vinszent** [comment]
I'm currently restructuring my code to use GError wherever possible and display them to the user. Also I just implemented empty states for all "containers" on the dev branch. Didn't really know how to use GError when I started GT so I just ignored them :P.

As for the actual error you got, `with code '2'` corresponds to `SOUP_STATUS_CANT_RESOLVE`, so some kind of problem with internet connectivity or DNS I guess.
